### PR TITLE
build: keep only postgresql as supported database backend

### PR DESCRIPTION
**Pull request by pstorz**
#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)

### Checklist for the _reviewer_ of the PR (will be processed by the Bareos team)

- [x] merge the following PR also when merging: https://git.bareos.com/bareos/CD/pulls/37

##### General

- [x] PR name is meaningful
- [x] Purpose of the PR is understood
- [x] Separate commit for this PR in the CHANGELOG.md, PR number referenced is same
- [x] Commit descriptions are understandable and well formatted

##### Source code quality

- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)
- [x] Required documentation changes are present and part of the PR
- [x] `bareos-check-sources --since-merge` does not report any problems
- [x] `git status` should not report modifications in the source tree after building and testing

### CODE REVIEW

**Review by arogge — COMMENTED** [review]
This looks reasonable.
However, we will have to update the documentation to reflect these changes.

File: core/src/stored/bscan.cc [hunk 1]
```diff
@@ -140,8 +140,7 @@ static void usage()
   fprintf(
       stderr,
       _("Usage: bscan [ options ] <device-name>\n"
-        "       -B <drivername>   specify the database driver name (default "
-        "NULL) <postgresql|mysql|sqlite3>\n"
+        "       -B                ignored for backwards compatibility.\n"
```
Comment (pstorz) [inline core/src/stored/bscan.cc:143]:
```suggestion
        "       -B                exists for backwards compatibility and is ignored.\n"
```

File: debian/bareos-database-common.config.in [hunk 2]
```diff
@@ -81,22 +76,6 @@ if [ -r @scriptdir@/bareos-config-lib.sh ]; then
                         db_set bareos-database-common/pgsql/app-pass ""
                     fi
                     ;;
-                mysql)
```
Comment (pstorz) [inline debian/bareos-database-common.config.in:84]: Why not remove the whole case..esac block from line 76?

**Review by pstorz — COMMENTED** [review]
Just some small comments

File: debian/bareos-database-common.config.in (hunk 2 above)
Comment (arogge) [reply debian/bareos-database-common.config.in:84]: I don't know, you were the last one who touched that file :smiley:

**Review by pstorz — COMMENTED** [review]
Just some small proposals for changes.

File: docs/manuals/source/DeveloperGuide/catalog.rst [hunk 3]
```diff
@@ -9,43 +9,22 @@ services and as such is not targeted at end users but rather at
 developers and system administrators who want or need to know more of
 the working details of **Bareos**.
 
-Currently, we prefer the usage of PostgreSQL. Therefore our support
-for SQLite or other databases could be discontinued in the future.
+Currently, we only support PostgreSQL.
```
[the remark this answers is not in the document]
Comment (bruno-at-bareos) [reply docs/manuals/source/DeveloperGuide/catalog.rst:12]: We make the choice to only support Postgresql ?

[the remark this answers is not in the document]
Comment (arogge) [reply docs/manuals/source/DeveloperGuide/catalog.rst:12]: yes, we made that choice years ago. 
PostgreSQL has been preferred for ages, SQLite was for testing only and installing new MySQL catalogs has been disabled in Bareos 20 last year.

**Review by arogge — APPROVED** [review]

**Pull request merged**